brizo/routes.py

```diff
@@ -262,39 +411,163 @@ def execute_compute_job():
+        type: string
+      - name: serviceAgreementId
+        in: query
+        description: The ID of the service agreement.
+        required: true
+        type: string
+      - name: consumerAddress
+        in: query
+        description: The consumer address.
+        required: true
+        type: string
+      - name: algorithmDID
+        in: query
+        description: hex str the did of the algorithm to be executed
+        required: false
+        type: string
+      - name: algorithmMeta
+        in: query
+        description: json object that define the algorithm attributes and url or raw code
+        required: false
+        type: string
+    responses:
+      200:
+        description: Call to the operator-service was successful.
+      400:
+        description: One of the required attributes is missing.
+      401:
+        description: Invalid asset data.
+      500:
+        description: Error
+    """
+    data = request.args
+    required_attributes = [
+        'signature',
+        'serviceAgreementId'
+    ]
+    msg, status = check_required_attributes(required_attributes, data, 'compute')
+    if msg:
+        return msg, status
+    if not (data.get('signature')):
+        return f'`signature is required in the call to "consume".', 400
+    # TODO  - check incoming signature
+    try:
+        workflow = dict()
+        workflow['workflow']=dict()
+        workflow['workflow']['agreementId']=data.get("serviceAgreementId")
+        workflow['workflow']['owner']=data.get("consumerAddress")
+        workflow['workflow']['stages']=list()
+        #build a new stage
+        stage = dict()
+        stage['index']=0
+        stage['input']=list()
+        stage['compute']=dict()
+        stage['algorithm']=dict()
+        stage['output']=dict()
+        #input props
+        inputs=dict()
+        inputs['index']=0
+        asset = keeper_instance().agreement_manager.get_agreement(data.get("serviceAgreementId"))
+        did = id_to_did(asset.did)
+        inputs['id']=did
+        inputs['url']=get_asset_urls(asset, provider_acc)
+        if not inputs['url']:
+            #there are no urls ??
+            return f'`cannot get url(s) in input did.', 400
+        stage['input'].append(inputs)
+        #compute prop
+        stage['compute']['Instances']=1
+        stage['compute']['namespace']="ocean-compute"
+        stage['compute']['maxtime']=3600
+        #algorithm prop
+        if data.get("algorithmDID") is None:
+          # use the metadata provided
+          algo=json.loads(data.get('algorithmMeta'))
+          stage['algorithm']['url']=algo.url
+          stage['algorithm']['rawcode']=algo.rawcode
+          stage['algorithm']['container']=dict()
+          stage['algorithm']['container']['image']=algo.container_image
+          stage['algorithm']['container']['tag']=algo.container_tag
+          stage['algorithm']['container']['entrypoint']=algo.container_entry_point
+        else:
+          # use the DID
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [indentation is not a multiple of four (comment)](https://app.codacy.com/manual/ocean-protocol/brizo/pullRequest?prid=4854028)